test/routes/add-article.test.ts

```diff
@@ -23,10 +23,29 @@ describe('add article', (): void => {
 
     expect(response.status).toBe(204);
     expect(await articles.count()).toBe(1);
-    expect([...articles][0]['http://schema.org/name']).toEqual('Article');
+    expect([...articles][0]['http://schema.org/name']).toEqual([{ '@value': 'Article' }]);
   });
 
   it('should throw an error if id is already set', async (): Promise<void> => {
     await expect(makeRequest(createArticle('_:1'))).rejects.toThrow(new createHttpError.Forbidden('Article IDs must not be set (\'_:1\' was given)'));
   });
+
+  it('should throw an error if it is not a schema:Article', async (): Promise<void> => {
+    await expect(makeRequest({
+      ...createArticle(),
+      '@type': 'http://schema.org/NewsArticle',
+    })).rejects.toThrow(new createHttpError.BadRequest('Article type must be http://schema.org/Article (\'http://schema.org/NewsArticle\' was given)'));
+  });
+
+  it.each([
+    [undefined],
+    [[]],
+    [{ '@value': null }],
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
Not ideal. No type, or `unknown` doesn't work either.

Reply:
Turns out it was the use of arrays that stopped `unknown` working (https://github.com/libero/article-store/pull/88/commits/9128fa48158c8efa0a39ab1be915a5538dfdcf09).